treewide: migrate packages to by-name

AndersonTorres opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Since `Hash-340235` manual migration is no longer discouraged.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

wolfgangwalther reviewed (approved):
Counted 16 packages removed in all-packages.nix and 16 .nix files renamed. Checked no files left over in the source directories.

LGTM.

AndersonTorres committed:
treewide: migrate packages maintained by AndersonTorres to by-name

Manual migration for the sake of by-name migration is no longer discouraged
since #340235.

Aleksanaa commented:
```
Nothing to be built.

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/341412

```

thiagokokada commented on pkgs/top-level/all-packages.nix:
```diff
@@ -429,8 +425,6 @@ with pkgs;
   };
 
   cereal = cereal_1_3_0;
```
Move this to `aliases.nix` maybe?

Aleksanaa replied:
Same as below

AndersonTorres replied:
This is a "green alias", because some packages don't work with the newer version.

thiagokokada commented on pkgs/top-level/all-packages.nix:
```diff
@@ -5100,22 +5055,8 @@ with pkgs;
 
   # Keep the old PGF since some documents don't render properly with
   # the new one.
```
This comment does't make sense anymore.

AndersonTorres replied:
Removed.

thiagokokada commented on pkgs/top-level/all-packages.nix:
```diff
@@ -5100,22 +5055,8 @@ with pkgs;
-
-  pgf3 = callPackage ../tools/typesetting/tex/pgf-tikz/pgf-3.x.nix { };
-
   pgf = pgf2;
```
Move this to `aliases.nix`?

Aleksanaa replied:
`aliases.nix` is for deprecation aliases.

https://github.com/NixOS/nixpkgs/blob/d52633a24c681fb4737664d5173c25bf330b244f/pkgs/top-level/aliases.nix#L3-L18

AndersonTorres committed:
pgf: remove outdated commentary

Aleksanaa reviewed (approved):
```
Nothing to be built.

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/341412
```

Aleksanaa commented:
Oh I should have edited the PR title to avoid making a big news
